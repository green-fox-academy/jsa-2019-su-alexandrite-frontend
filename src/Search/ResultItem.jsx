import React from 'react';
import PropTypes from 'prop-types';
import {
  View,
  Text,
  Image,
} from 'react-native';
import { useNavigation } from 'react-navigation-hooks';
import { TouchableHighlight } from 'react-native-gesture-handler';
import styles from './styles';

import tradeIcon from '../../assets/icons/search-page/trade.png';
import detailIcon from '../../assets/icons/search-page/detail.png';

const ResultItem = ({ symbol, exchange }) => {
  const { push } = useNavigation();
  return (
    <View style={styles.resultList}>
      <View>
        <Text style={styles.stockName}>{symbol}</Text>
        <Text style={styles.stockExchange}>{exchange}</Text>
      </View>
      <View style={styles.resultButton}>
        <TouchableHighlight>
          <Image source={tradeIcon} />
        </TouchableHighlight>
        <TouchableHighlight
          style={styles.detailButton}
          onPress={() => push('StockDetails', { symbol })}
          activeOpacity={0.5}
          underlayColor="#0000"
        >
          <Image source={detailIcon} />
        </TouchableHighlight>
      </View>
    </View>
  );
};

ResultItem.propTypes = {
  symbol: PropTypes.string.isRequired,
  exchange: PropTypes.string.isRequired,
};

export default ResultItem;
