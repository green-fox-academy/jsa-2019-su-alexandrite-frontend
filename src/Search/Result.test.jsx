import React from 'react';
import { useSelector } from 'react-redux';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Result from './Result';
import ResultItem from './ResultItem';

jest.mock('react-redux');

const SAMPLE_STATE = {
  search: {
    isLoading: false,
    result: [{
      symbol: 'AAPL',
      exchange: 'asdasd',
    }],
    error: null,
    touched: true,
  },
};
describe('<Result />', () => {
  beforeEach(() => {
    useSelector.mockReturnValue(SAMPLE_STATE);
  });

  afterAll(() => {
    jest.resetModules();
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    const wrapper = shallow(<Result />);
    expect(wrapper).not.toBeNull();
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const tree = renderer.create(
      <Result />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render and match the snapshot when touched and no result available', () => {
    useSelector.mockReturnValueOnce({ ...SAMPLE_STATE, touched: true, result: [] });
    const tree = renderer.create(
      <Result />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render and match the snapshot when not touched and no result available', () => {
    useSelector.mockReturnValueOnce({ ...SAMPLE_STATE, touched: false, result: [] });
    const tree = renderer.create(
      <Result />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render and match the snapshot when loading', () => {
    useSelector.mockReturnValueOnce({ ...SAMPLE_STATE, isLoading: true });
    const tree = renderer.create(
      <Result />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Should render and match the snapshot on error', () => {
    useSelector.mockReturnValueOnce({ ...SAMPLE_STATE, error: { message: 'sample message' } });
    const tree = renderer.create(
      <Result />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('FlatList key extractor should get the symbol field of the data elements', () => {
    const wrapper = shallow(
      <Result />,
    );
    const sampleData = {
      symbol: 'AAPL',
    };
    const { keyExtractor } = wrapper.find('FlatList').props();
    expect(keyExtractor(sampleData)).toEqual(sampleData.symbol);
  });

  it('FlatList renderItem should render resultItem', () => {
    const wrapper = shallow(
      <Result />,
    );
    const sampleData = SAMPLE_STATE.search.result[0];
    const { renderItem } = wrapper.find('FlatList').props();
    expect(renderItem({ item: sampleData }))
      .toEqual(<ResultItem symbol={sampleData.symbol} exchange={sampleData.exchange} />);
  });
});
