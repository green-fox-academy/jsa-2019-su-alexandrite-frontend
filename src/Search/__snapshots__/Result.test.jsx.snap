// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Result /> Should render and match the snapshot 1`] = `
<View>
  <RCTScrollView
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`<Result /> Should render and match the snapshot on error 1`] = `
<View>
  <Text>
    sample message
  </Text>
</View>
`;

exports[`<Result /> Should render and match the snapshot when loading 1`] = `
<View>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="large"
    style={
      Object {
        "flex": 1,
        "margin": 40,
      }
    }
  />
</View>
`;

exports[`<Result /> Should render and match the snapshot when not touched and no result available 1`] = `
<View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;

exports[`<Result /> Should render and match the snapshot when touched and no result available 1`] = `
<Text
  style={
    Object {
      "marginTop": 20,
      "textAlign": "center",
    }
  }
>
  No Result
</Text>
`;
